libraries/shared/factories/storage/keys_lookup_test.go

```diff
@@ -53,53 +55,59 @@ var _ = Describe("Storage keys lookup", func() {
 			It("returns error if refreshing keys fails", func() {
 				loader.LoadMappingsError = fakes.FakeError
 
-				_, err := lookup.Lookup(fakes.FakeHash)
+				_, err := lookup.Lookup(test_data.FakeHash())
 
 				Expect(err).To(HaveOccurred())
 				Expect(err).To(MatchError(fakes.FakeError))
 			})
 		})
 
 		Describe("when key found", func() {
+			var fakeKey common.Hash
+
 			BeforeEach(func() {
-				loader.StorageKeyMappings = map[common.Hash]types.ValueMetadata{fakes.FakeHash: fakeMetadata}
-				_, err := lookup.Lookup(fakes.FakeHash)
+				fakeKey = test_data.FakeHash()
+				loader.StorageKeyMappings = map[common.Hash]types.ValueMetadata{fakeKey: fakeMetadata}
+				_, err := lookup.Lookup(fakeKey)
 				Expect(err).NotTo(HaveOccurred())
```

Comment:
Are these expects necessary? They'll fail on lines 60 and 61 if this is broken.

Reply:
Hm are you talking about:
```
				Expect(err).To(HaveOccurred())
				Expect(err).To(MatchError(fakes.FakeError))
```
I think the reasons that test would fail are different than the reasons that these expectations would fail.

I do think this test is probably due for some cleanup, though. From what I can glean, it seems like the expectations on lines 79-80 are verifying that a subsequent call to `Lookup` doesn't trigger mappings being loaded after they have already been loaded once for a given key - so the expectation in the `BeforeEach` verifies that the call count is 1 and the expectation in the test verifies that it's still 1. If we removed the expectation in the `BeforeEach`, then it's possible that the expectation in the test could pass even though the call to `LoadMappings` did happen. Does that make sense?

Reply:
It mostly does - it looks like you're aiming for that Rails helper "should_be_changed_by" - something like that I don't remember verbatim.

What I'd say I mildly dislike here is the BeforeEach when there is only one `It` block. After all it could be just one test. #notablocker

Reply:
Yeah, totally agreed. I think we may have gotten a little over-aggressive with the `BeforeEach` stuff 😅